tests/unit/utils/test_filter_messages.py

```diff
@@ -0,0 +1,163 @@
+from langchain_core.messages import AIMessage, HumanMessage, SystemMessage
+
+from agents.common.constants import FINALIZER
+from utils.filter_messages import (
+    filter_messages_via_checks,
+    filter_most_recent_messages,
+    is_finalizer_message,
+    is_human_message,
+    is_system_message,
+)
+
+
+@pytest.mark.parametrize(
+    "test_case,messages,checks,expected_result",
+    [
+        (
+            "should return no messages when no checks are applied",
+            [
+                HumanMessage(content="Human Message"),
+                AIMessage(content="AI Message"),
+                SystemMessage(content="Sysytem Message"),
+            ],
+            [],  # no checks
+            [],
+        ),
+        (
+            "should return only human messages",
+            [
+                HumanMessage(content="Human Message"),
+                AIMessage(content="AI Message"),
+                SystemMessage(content="Sysytem Message"),
+            ],
+            [is_human_message],
+            [
+                HumanMessage(content="Human Message"),
+            ],
+        ),
+        (
+            "should return only finalizer messages",
+            [
+                HumanMessage(content="Human Message"),
+                AIMessage(content="AI Message"),
+                # Finalizer messages are AIMessages with the name "FINALIZER":
+                AIMessage(content="Finalizer Message", name=FINALIZER),
+                SystemMessage(content="Sysytem Message"),
+            ],
+            [is_finalizer_message],
+            [
+                AIMessage(content="Finalizer Message", name=FINALIZER),
+            ],
+        ),
+        (
+            "should return only system messages",
+            [
+                HumanMessage(content="Human Message"),
+                AIMessage(content="AI Message"),
+                SystemMessage(content="Sysytem Message"),
+            ],
+            [is_system_message],
+            [
+                SystemMessage(content="Sysytem Message"),
+            ],
+        ),
+        (
+            "should handle multiple checks",
+            [
+                HumanMessage(content="Human Message"),
+                AIMessage(content="AI Message"),
+                AIMessage(content="Finalizer Message", name=FINALIZER),
+                SystemMessage(content="Sysytem Message"),
+            ],
+            [is_human_message, is_finalizer_message],
+            [
+                HumanMessage(content="Human Message"),
+                AIMessage(content="Finalizer Message", name=FINALIZER),
+            ],
+        ),
+        (
+            "should return empty list when no messages pass the checks",
+            [
+                HumanMessage(content="Human Message"),
+                AIMessage(content="AI Message"),
+                SystemMessage(content="Sysytem Message"),
+            ],
+            [is_finalizer_message],
+            [],
+        ),
+    ],
```

Comment:
can we also add test case where all three filters are considered? 
`[is_human_message, is_finalizer_message, is_system_message]`

Reply:
sure